### DISCUSSION TITLE
Inheritance and locality of behaviour

### DISCUSSION THREAD

**Discussion by TimCanu** [bigskysoftware/htmx#2835] [General]
Hello, :slightly_smiling_face: 

I feel like having HTMX using inheritance by default goes against the locality of behaviour principle. Having a `hx-boost="true"` at a high level in the DOM to encapsulate all links seems smart, but you it is a double edged sword. I had some cases in which I spent a bit of time figuring out why stuff was not working, and it was related to inheritance (which is not obvious to me). So I end up adding `hx-disinherit="*"`.
I have a strong feeling that inheriting should not be the default behaviour.

**Comment by Telroshan** [comment]
Hey, the inheritance topic is a recurring one; it has been discussed a few times on the discord, and also in issues like #1520 .
I also disable inheritance for my own projects, but others prefer it this way. Note that changing this default would be a breaking change, thus would require a major version upgrade, which is very unlikely to happen anytime soon, if at all.

Please note that, on top of `hx-disinherit` that you can define on specific elements, you can also toggle `htmx.config.disableInheritance`, which I realize as I'm writing this, is still undocumented in the website's [configuration reference](https://htmx.org/reference/#config), oops 😅 
You can read its mention in the [attributes inheritance](https://htmx.org/docs/#inheritance) docs though.

**Reply by TimCanu** [reply]
Hey. I did not realize that it could be disabled through the config. This is a good solution :slightly_smiling_face: 
If in the future it happens that htmx has to do a major version upgrade, I hope that making inheritance an extension will be considered!
Thank you for your answer and for your work on htmx. :wink:
